diff --git a/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaDataUtils.kt b/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaDataUtils.kt
index aed654fd..0282c846 100644
--- a/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaDataUtils.kt
+++ b/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaDataUtils.kt
@@ -12,6 +12,7 @@
 package com.adobe.marketing.mobile.messaging
 
 import androidx.annotation.VisibleForTesting
+import androidx.compose.ui.Alignment
 import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepButton
 import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepIcon
 import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepImage
@@ -40,15 +41,13 @@ internal object ContentCardSchemaDataUtils {
      * @return The built [AepUITemplate] or null if the proposition is not a content card or parsing fails.
      */
     internal fun buildTemplate(proposition: Proposition): AepUITemplate? {
-
         if (!isContentCard(proposition)) return null
 
-        if (proposition.items.size <= 0) return null
-
         val propositionItem = proposition.items[0]
+        val isRead = propositionItem.itemData.getValue("read") as? Boolean ?: false
 
         val baseTemplateModel: AepUITemplate? = propositionItem.contentCardSchemaData?.let {
-            getTemplate(it)
+            getTemplate(it, isRead)
         }
         return baseTemplateModel
     }
@@ -60,7 +59,7 @@ internal object ContentCardSchemaDataUtils {
      * @return The parsed [AepUITemplate] or null if parsing fails.
      */
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun getTemplate(contentCardSchemaData: ContentCardSchemaData): AepUITemplate? {
+    internal fun getTemplate(contentCardSchemaData: ContentCardSchemaData, isRead: Boolean): AepUITemplate? {
         try {
             val contentMap =
                 contentCardSchemaData.content as? Map<String, Any>
@@ -76,15 +75,15 @@ internal object ContentCardSchemaDataUtils {
 
                 when (templateType) {
                     AepUITemplateType.SMALL_IMAGE.typeName -> {
-                        return createSmallImageTemplate(contentMap, id)
+                        return createSmallImageTemplate(contentMap, id, isRead)
                     }
 
                     AepUITemplateType.LARGE_IMAGE.typeName -> {
-                        return createLargeImageTemplate(contentMap, id)
+                        return createLargeImageTemplate(contentMap, id, isRead)
                     }
 
                     AepUITemplateType.IMAGE_ONLY.typeName -> {
-                        return createImageOnlyTemplate(contentMap, id)
+                        return createImageOnlyTemplate(contentMap, id, isRead)
                     }
 
                     else -> {
@@ -107,9 +106,20 @@ internal object ContentCardSchemaDataUtils {
         }
     }
 
+    internal fun getAlignmentFromString(align: String?): Alignment {
+        return when (align) {
+            MessagingConstants.ContentCard.UIKeys.TOP_LEFT -> Alignment.TopStart
+            MessagingConstants.ContentCard.UIKeys.TOP_RIGHT -> Alignment.TopEnd
+            MessagingConstants.ContentCard.UIKeys.BOTTOM_LEFT -> Alignment.BottomStart
+            MessagingConstants.ContentCard.UIKeys.BOTTOM_RIGHT -> Alignment.BottomEnd
+            else -> Alignment.TopStart
+        }
+    }
+
     private fun createSmallImageTemplate(
         contentMap: Map<String, Any>,
-        cardId: String
+        cardId: String,
+        isRead: Boolean
     ): SmallImageTemplate? {
         val title = createAepText(contentMap, MessagingConstants.ContentCard.UIKeys.TITLE, cardId)
         if (title == null || title.content.isEmpty()) {
@@ -127,13 +137,15 @@ internal object ContentCardSchemaDataUtils {
             image = createAepImage(contentMap, cardId),
             actionUrl = getActionUrl(contentMap, cardId),
             buttons = createAepButtonsList(contentMap, cardId),
-            dismissBtn = createAepDismissButton(contentMap, cardId)
+            dismissBtn = createAepDismissButton(contentMap, cardId),
+            isRead = isRead
         )
     }
 
     private fun createLargeImageTemplate(
         contentMap: Map<String, Any>,
-        cardId: String
+        cardId: String,
+        isRead: Boolean
     ): LargeImageTemplate? {
         val title = createAepText(contentMap, MessagingConstants.ContentCard.UIKeys.TITLE, cardId)
         if (title == null || title.content.isEmpty()) {
@@ -151,13 +163,15 @@ internal object ContentCardSchemaDataUtils {
             image = createAepImage(contentMap, cardId),
             actionUrl = getActionUrl(contentMap, cardId),
             buttons = createAepButtonsList(contentMap, cardId),
-            dismissBtn = createAepDismissButton(contentMap, cardId)
+            dismissBtn = createAepDismissButton(contentMap, cardId),
+            isRead = isRead
         )
     }
 
     private fun createImageOnlyTemplate(
         contentMap: Map<String, Any>,
-        cardId: String
+        cardId: String,
+        isRead: Boolean
     ): ImageOnlyTemplate? {
         val image = createAepImage(contentMap, cardId)
         if (image == null || image.url.isNullOrBlank()) {
@@ -172,7 +186,8 @@ internal object ContentCardSchemaDataUtils {
             id = cardId,
             image = image,
             actionUrl = getActionUrl(contentMap, cardId),
-            dismissBtn = createAepDismissButton(contentMap, cardId)
+            dismissBtn = createAepDismissButton(contentMap, cardId),
+            isRead = isRead
         )
     }
 
diff --git a/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/EdgePersonalizationResponseHandler.java b/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/EdgePersonalizationResponseHandler.java
index 7fa5c095..c9719e96 100644
--- a/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/EdgePersonalizationResponseHandler.java
+++ b/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/EdgePersonalizationResponseHandler.java
@@ -521,6 +521,7 @@ class EdgePersonalizationResponseHandler {
         for (final Map.Entry<Surface, List<Proposition>> propositionEntry :
                 requestedPropositions.entrySet()) {
             for (final Proposition proposition : propositionEntry.getValue()) {
+                InternalMessagingUtils.updateReadStatus(proposition, parent.getApi());
                 convertedPropositions.add(proposition.toEventData());
             }
         }
diff --git a/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InternalMessagingUtils.java b/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InternalMessagingUtils.java
index d67beb6d..6a65edf3 100644
--- a/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InternalMessagingUtils.java
+++ b/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InternalMessagingUtils.java
@@ -13,11 +13,16 @@ package com.adobe.marketing.mobile.messaging;
 
 import androidx.annotation.Nullable;
 import androidx.annotation.VisibleForTesting;
+
+import com.adobe.marketing.mobile.AdobeCallbackWithError;
 import com.adobe.marketing.mobile.AdobeError;
 import com.adobe.marketing.mobile.Event;
+import com.adobe.marketing.mobile.EventHistoryRequest;
+import com.adobe.marketing.mobile.EventHistoryResult;
 import com.adobe.marketing.mobile.EventSource;
 import com.adobe.marketing.mobile.EventType;
 import com.adobe.marketing.mobile.ExtensionApi;
+import com.adobe.marketing.mobile.MessagingEdgeEventType;
 import com.adobe.marketing.mobile.launch.rulesengine.LaunchRule;
 import com.adobe.marketing.mobile.services.DataStoring;
 import com.adobe.marketing.mobile.services.DeviceInforming;
@@ -32,6 +37,9 @@ import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
@@ -695,6 +703,44 @@ class InternalMessagingUtils {
         return updatedMap;
     }
 
+    static void updateReadStatus(final Proposition proposition, final ExtensionApi extensionApi) {
+        if (proposition == null || proposition.getItems().isEmpty()) {
+            return;
+        }
+        String activityId = proposition.getActivityId();
+        if (StringUtils.isNullOrEmpty(activityId)) {
+            return;
+        }
+        final boolean[] isRead = {false};
+        final Map<String, Object> eventData = new HashMap<>();
+        eventData.put(MessagingConstants.EventMask.Mask.EVENT_TYPE, MessagingEdgeEventType.INTERACT.getPropositionEventType());
+        eventData.put(MessagingConstants.EventMask.Mask.ACTIVITY_ID, activityId);
+        EventHistoryRequest[] eventHistoryRequest = {new EventHistoryRequest(eventData, 0, 0)};
+        try {
+            CountDownLatch latch = new CountDownLatch(1);
+            extensionApi.getHistoricalEvents(eventHistoryRequest, false, new AdobeCallbackWithError<EventHistoryResult[]>() {
+                @Override
+                public void fail(AdobeError error) {
+                    latch.countDown();
+                }
+
+                @Override
+                public void call(EventHistoryResult[] value) {
+                    if (value != null) {
+                        if (value.length > 0 && value[0] != null) {
+                            isRead[0] = value[0].count > 0;
+                        }
+                    }
+                    latch.countDown();
+                }
+            });
+            latch.await(500, TimeUnit.MILLISECONDS);
+            proposition.getItems().get(0).getItemData().put("read", isRead[0]);
+        } catch (InterruptedException e) {
+            Log.debug(MessagingConstants.LOG_TAG, SELF_TAG, "updateReadStatus - failed to get historical events for proposition %s: %s", activityId, e.getLocalizedMessage());
+        }
+    }
+
     @VisibleForTesting
     static void resetPushTokenSyncTimestamp() {
         lastPushTokenSyncTimestamp = 0;
